### PR TITLE
AEROGEAR-1997 Intergrate Security Service

### PR DESCRIPTION
**Pull request by aidenkeating**
### Motivation
WIP to add AeroGear Security Service.

### Progress
PRs backing these features are still open.

- [x] Lock Screen Check
- [x] Rooted Check
- [x] Emulator Check
- [x] Debugger Check
- [x] Developer Options Check

Resolves https://github.com/feedhenry/mobile-security-android-template/issues/39 , I think

### CODE REVIEW

**Comment by wei-lee** [comment]
LGTM

**Comment by tomjackman** [comment]
Just a side note. Since the asciidoc tags have been renamed, we will need to update the mobile-security docs too - https://github.com/feedhenry/mobile-security/blob/master/docs/mobileAppSelfDefenceOverview.adoc

**Comment by tomjackman** [comment]
@aidenkeating Is there any more dev work required here? If not, I can update the docs [here](https://github.com/feedhenry/mobile-security/blob/master/docs/mobileAppSelfDefenceOverview.adoc) and merge this when it's ready to go.

**Comment by aidenkeating** [comment]
@TommyJ1994 There is a small amount, just adding the final checks in now

File: app/src/main/java/com/feedhenry/securenativeandroidtemplate/features/device/DeviceFragment.java [hunk 1]
```diff
@@ -263,14 +243,9 @@ public boolean checkAppInstalled(String packageName) {
...
-            PackageInfo packageInfo;
-            packageInfo = context.getPackageManager().getPackageInfo(context.getPackageName(), 0);
-            if ((packageInfo.applicationInfo.flags & ApplicationInfo.FLAG_ALLOW_BACKUP) != 0) {
-                setDetected(allowBackup, R.string.allow_backup_detected_positive);
-            }
-        } catch (PackageManager.NameNotFoundException e) {
-            e.printStackTrace();
+        SecurityCheckResult result = securityService.check(SecurityCheckType.ALLOW_BACKUP_ENABLED);
+        if (result.passed()) {
+            setDetected(allowBackup, R.string.allow_backup_detected_positive);
```
Comment (aidenkeating) [inline app/src/main/java/com/feedhenry/securenativeandroidtemplate/features/device/DeviceFragment.java:248]: @TommyJ1994 Quick question from curiosity. Is this the right way around (taken from the SDK example app). If backup is enabled that's a bad thing? I thought it'd be a neutral thing?

Comment (tomjackman) [reply app/src/main/java/com/feedhenry/securenativeandroidtemplate/features/device/DeviceFragment.java:248]: @aidenkeating  Yes, its a bad thing in the sense that you can retrieve application data without root access.

Comment (aidenkeating) [reply app/src/main/java/com/feedhenry/securenativeandroidtemplate/features/device/DeviceFragment.java:248]: @TommyJ1994 Found information in https://github.com/feedhenry/mobile-security, very useful to know, thanks

File: app/src/androidTest/java/com/feedhenry/securenativeandroidtemplate/di/SecureApplicationTestModule.java [hunk 2]
```diff
@@ -86,6 +87,11 @@ MobileCore provideMobileCore(Context context) {
         return mobileCore;
     }
 
+    @Provides @Singleton
```
Comment (tomjackman) [inline app/src/androidTest/java/com/feedhenry/securenativeandroidtemplate/di/SecureApplicationTestModule.java:90]: Hi @aidenkeating. Is there any chance you could add an asciidoc tag to this function? I just want to add some docs for where the security service is being initialized etc. eg.

```
// tag::securityServiceInit[]
@Provides @Singleton
SecurityService securityService(Context context, MobileCore mobileCore) {
   return mobileCore.getInstance(SecurityService.class);
}
// end::securityServiceInit[]
```

Another example is [here](https://github.com/aidenkeating/mobile-security-android-template/blob/092e554020a54e2b8ea97a4592dd0c914f77b66b/app/src/main/java/com/feedhenry/securenativeandroidtemplate/di/SecureApplicationModule.java#L113-L124)

**Comment by tomjackman** [comment]
@aidenkeating I have the docs ready to go locally I think now, so this is fine to be merged when its ready. I'll regenerate the html with include these changes and open a PR in https://github.com/feedhenry/mobile-security then.

File: app/src/androidTest/java/com/feedhenry/securenativeandroidtemplate/di/SecureApplicationTestModule.java (hunk 2 above)
Comment (aidenkeating) [reply app/src/androidTest/java/com/feedhenry/securenativeandroidtemplate/di/SecureApplicationTestModule.java:90]: @TommyJ1994 Just to confirm add tags to provideSecurityService in SecurityApplicationTestModule or SecurityApplicationModule?

**Comment by aidenkeating** [comment]
@TommyJ1994 Added tag. Not sure why the tests are failing, they're passing locally and seem to fail at different parts of the the notes test on CircleCI each run.

File: app/src/androidTest/java/com/feedhenry/securenativeandroidtemplate/di/SecureApplicationTestModule.java (hunk 2 above)
Comment (tomjackman) [reply app/src/androidTest/java/com/feedhenry/securenativeandroidtemplate/di/SecureApplicationTestModule.java:90]: Just the `SecurityApplicationModule` is fine.

**Comment by tomjackman** [comment]
@aidenkeating Thanks. I can probably take a look at fixing this later. Since the failing test is unrelated to the changes in this PR, I think we can ignore this for now. I'm seeing similar issues with CircleCI though.

**Pull request merged**